static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -2,179 +2,233 @@ define([
     'common/utils/fastdom-promise',
     'Promise',
     'common/utils/$',
+    'common/utils/create-store',
     'common/utils/mediator',
     'common/utils/config',
-    'common/utils/detect'
+    'common/utils/detect',
+    'lodash/objects/assign'
 ], function (
     fastdom,
     Promise,
     $,
+    createStore,
     mediator,
     config,
-    detect) {
+    detect,
+    assign) {
+
+    var $adBanner = $('.top-banner-ad-container--above-nav');
+    var $adBannerInner = $('.ad-slot--top-above-nav', $adBanner);
+    var $header = $('.js-header');
+    var $document = $(document.body);
+
+    var getClientAdHeight = function () {
+        return fastdom.read(function () {
+            return $adBannerInner[0].clientHeight;
+        });
+    };
 
-    return function () {
-        if (detect.getBreakpoint() !== 'mobile' && config.page.contentType !== 'Interactive') {
-            fastdom.read(function () {
-                var adId = 'dfp-ad--top-above-nav',
-                    $adBanner = $('.js-top-banner-above-nav'),
-                    $adBannerInner = $('#' + adId, $adBanner),
-                    $header = $('.js-header');
-
-                var getClientAdHeight = function () {
-                    return fastdom.read(function () {
-                        return $adBannerInner[0].clientHeight;
+    var getAdIframe = function () { return $('iframe', $adBanner); };
+
+    var topAdRenderedPromise = new Promise(function (resolve) {
```

Comment:
That looks perfect. <3

So I just need to do:

``` js
var topAdRenderedPromise = trackAd(dfpAdSlotId);
```

It looks like this will return `Promise[Boolean]`. What does the boolean represent?